### DISCUSSION TITLE
Conditional table cell - text color

### DISCUSSION THREAD

**Discussion by melohagan** [Budibase/budibase#10204] [Show and tell]
As this feature has been requested a few times, I thought it would be worth adding a tutorial on how to implement this using the wonderful **Super Table** component by @poirazis

A common use-case is to highlight cells in a different color depending on a status value.

To begin using it, follow the quick guide on [Importing plugins](https://docs.budibase.com/docs/custom-plugin#importing-plugins).

The **Super Table** is a multi-part custom component plugin requiring the following imports:
 - [Super Table](https://github.com/poirazis/bb-component-SuperTable)
 - [Super Column](https://github.com/poirazis/bb-component-SuperTableColumn) 
 - [Super Cell](https://github.com/poirazis/bb-component-SuperTableCell)

Once those components have been imported, create a new app with the sample data. Add a screen with a **Data provider** for the *Inventory* table.

Under the **Data provider**, nest the **Super Table**, and then nest a **Super Column** for each of the table columns you want to display. 

Under the **Super Column** that was added for the *Status* column, nest a **Super Cell**. In the *Conditions* tab of the cell, click `Configure conditions` and setup the following, using the duplicate icon button to speed things up. Make sure to click `Save`!

![Screenshot 2023-04-04 at 13 50 52](https://user-images.githubusercontent.com/101575380/229797076-16b8d28d-a3d4-4553-a728-7de3162f02fe.png)

The binding in each case comes from the *Status column* binding:

![Screenshot 2023-04-04 at 13 52 35](https://user-images.githubusercontent.com/101575380/229797549-e0063e16-a30c-4396-8bcc-8d55b09c0382.png)

#### Result

![Screenshot 2023-04-04 at 13 50 28](https://user-images.githubusercontent.com/101575380/229796990-6b003e45-a665-49c3-a9e2-07607f8abb29.png)

**Comment by xnohat** [comment]
Superb plugins , that all we need <3

**Comment by jo-chla** [comment]
Hi, I have installed the Super Table/Column/Cell plugins, but was wondering is there a way to have the Font Color of a cell change if the value of that cell already exists in the column? 

For example, if "Name" column contained "Jeff, Sam, Mike, Mike, Jeff, Jeff", the cells with "Jeff" and "Mike" would all be a certain color.

**Reply by melohagan** [reply]
Hey @jo-chla You may be able to do this with a Data provider. This approach would be limited to checking the names within the limit set by the data provider.

But then your *Update setting* condition would use something like: IF "Mike" **Is In** Binding `{{ pluck Data provider.rows 'Name' }}`

**Reply by jo-chla** [reply]
Thanks! However I have tried with a numerical column:
IF **{{ New Super Table Column.Cell Value }}** IS IN Binding **{{ pluck Data provider.rows 'ID_Number' }}**
And it is not updating the color of entries that already exist in ID_Number column

Another question would it be better to do in the budibase backend database interface which is much faster? 
I have tried creating a function column in MyTable:
**{{ #inArray ( pluck MyTable.rows 'ID_Number' ) ID_Number }}** Duplicate **{{/inArray}}**
but it hasn't been working

**Comment by cyberz0mbie** [comment]
Does this work with the latest version of budibase, I set mine up to your example and i do not get anything on my screen

**Comment by ahmedelnaqa** [comment]
hi,
I try to highlight "change row background colour" based on the date of DateColumn=today

![image](https://github.com/Budibase/budibase/assets/72502706/d2f2d404-2554-4754-83de-b49ed3a84fa7)
thanks

**Comment by Senbeats** [comment]
Followed the tutorial but im still having some issues! dont suppose you i can add you on discord or something? would really appreciate some help! 

Im trying to make it so a date column will change color based on the current date -

**Comment by vusd-adam** [comment]
Maybe a dumb question, but when I add a super table, it adds all the columns for that table by default. Then when I add each of the super columns, it adds each column again. So I end up with two sets of columns, the columns in the super table and also the super columns. Is there a way to hide all the super table columns or something?

**Reply by vikinim** [reply]
same with me . unable to understand what to do with 2 sets of column. ( duplicate column )

**Comment by vikinim** [comment]
when i am adding super table , then data provider adding columns and super table adding another set of columns . so how to hide columns added by data provider ?
